README.md

```diff
@@ -104,6 +105,99 @@ Renders command usage to a string value.
 
 Renders usage and prints to the console
 
+## Autocompletions
+
+Citty works with [tab](https://github.com/bombshell-dev/tab) to provide shell autocompletions across `zsh`, `bash`, `fish`, and `powershell`.
+
+```sh
+npm install @bomb.sh/tab
+```
+
+Add autocompletions to your citty CLI:
+
+```ts
+import { defineCommand, createMain } from "citty";
+import tab from "@bomb.sh/tab/citty";
+
+const main = defineCommand({
+  meta: {
+    name: "my-cli",
+    description: "My CLI tool",
+  },
+  subCommands: {
+    dev: defineCommand({
+      meta: {
+        name: "dev",
+        description: "Start dev server",
+      },
+      args: {
+        port: {
+          type: "string",
+          description: "Specify port",
+        },
+        host: {
+          type: "string",
+          description: "Specify host",
+        },
+      },
+    }),
+  },
+});
+
+const completion = await tab(main);
+
+// Add custom completions for option values
+const devCommand = completion.commands.get("dev");
+const portOption = devCommand.options.get("port");
```

Comment:
this is the only part that's a bit annoying that'd be solved if we integrate tab natively in citty.